File: csharp-dotnet2/src/main/CsharpDotNet2/Org/OpenAPITools/Model/Attachments.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace Org.OpenAPITools.Model {

  /// <summary>
  /// Dictionary containing details of a file uploaded by a user. 
  /// </summary>
  [DataContract]
  public class Attachments {
    /// <summary>
    /// The unique ID for the attachment. 
    /// </summary>
    /// <value>The unique ID for the attachment. </value>
    [DataMember(Name="id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "id")]
    public int? Id { get; set; }

    /// <summary>
    /// Name of the uploaded file. 
    /// </summary>
    /// <value>Name of the uploaded file. </value>
    [DataMember(Name="name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "name")]
    public string Name { get; set; }

    /// <summary>
    /// A representation of the path of the file within the repository of user-uploaded files.  If the `path_id` of a file is `{realm_id}/ab/cdef/temp_file.py`, its URL will be: `{server_url}/user_uploads/{realm_id}/ab/cdef/temp_file.py`. 
    /// </summary>
    /// <value>A representation of the path of the file within the repository of user-uploaded files.  If the `path_id` of a file is `{realm_id}/ab/cdef/temp_file.py`, its URL will be: `{server_url}/user_uploads/{realm_id}/ab/cdef/temp_file.py`. </value>
    [DataMember(Name="path_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "path_id")]
    public string PathId { get; set; }

    /// <summary>
    /// Size of the file in bytes. 
    /// </summary>
    /// <value>Size of the file in bytes. </value>
    [DataMember(Name="size", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "size")]
    public int? Size { get; set; }

    /// <summary>
    /// Time when the attachment was uploaded as a UNIX timestamp multiplied by 1000 (matching the format of getTime() in JavaScript).  **Changes**: Changed in Zulip 2.2 (feature level 22).  This field was previously a floating point number. 
    /// </summary>
    /// <value>Time when the attachment was uploaded as a UNIX timestamp multiplied by 1000 (matching the format of getTime() in JavaScript).  **Changes**: Changed in Zulip 2.2 (feature level 22).  This field was previously a floating point number. </value>
    [DataMember(Name="create_time", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "create_time")]
    public int? CreateTime { get; set; }

    /// <summary>
    /// Contains basic details on any Zulip messages that have been sent referencing this [uploaded file](/api/upload-file). This includes messages sent by any user in the Zulip organization who sent a message containing a link to the uploaded file. 
    /// </summary>
    /// <value>Contains basic details on any Zulip messages that have been sent referencing this [uploaded file](/api/upload-file). This includes messages sent by any user in the Zulip organization who sent a message containing a link to the uploaded file. </value>
    [DataMember(Name="messages", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "messages")]
    public List<AttachmentsMessages> Messages { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class Attachments {\n");
      sb.Append("  Id: ").Append(Id).Append("\n");
      sb.Append("  Name: ").Append(Name).Append("\n");
      sb.Append("  PathId: ").Append(PathId).Append("\n");
      sb.Append("  Size: ").Append(Size).Append("\n");
      sb.Append("  CreateTime: ").Append(CreateTime).Append("\n");
      sb.Append("  Messages: ").Append(Messages).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return Newtonsoft.Json.JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
    }

}
}

File: dart-jaguar/lib/model/attachments.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';


import 'package:openapi/model/attachments_messages.dart';

part 'attachments.jser.dart';

class Attachments {
   /* The unique ID for the attachment.  */
  @Alias('id', isNullable: false,  )
  final int id;
   /* Name of the uploaded file.  */
  @Alias('name', isNullable: false,  )
  final String name;
   /* A representation of the path of the file within the repository of user-uploaded files.  If the `path_id` of a file is `{realm_id}/ab/cdef/temp_file.py`, its URL will be: `{server_url}/user_uploads/{realm_id}/ab/cdef/temp_file.py`.  */
  @Alias('path_id', isNullable: false,  )
  final String pathId;
   /* Size of the file in bytes.  */
  @Alias('size', isNullable: false,  )
  final int size;
   /* Time when the attachment was uploaded as a UNIX timestamp multiplied by 1000 (matching the format of getTime() in JavaScript).  **Changes**: Changed in Zulip 2.2 (feature level 22).  This field was previously a floating point number.  */
  @Alias('create_time', isNullable: false,  )
  final int createTime;
   /* Contains basic details on any Zulip messages that have been sent referencing this [uploaded file](/api/upload-file). This includes messages sent by any user in the Zulip organization who sent a message containing a link to the uploaded file.  */
  @Alias('messages', isNullable: false,  )
  final List<AttachmentsMessages> messages;
  

  Attachments(
      

{
     this.id = null,  
     this.name = null,  
     this.pathId = null,  
     this.size = null,  
     this.createTime = null,  
     this.messages = const [] 
    
    }
  );

  @override
  String toString() {
    return 'Attachments[id=$id, name=$name, pathId=$pathId, size=$size, createTime=$createTime, messages=$messages, ]';
  }
}

@GenSerializer(nullableFields: true)
class AttachmentsSerializer extends Serializer<Attachments> with _$AttachmentsSerializer {

}

